treeherder/webapp/api/performance_serializers.py

```diff
@@ -116,8 +117,65 @@ class Meta:
         ]
 
 
+class AlertTaskclusterMetadataSerializer(serializers.ModelSerializer):
+    current_push = serializers.SerializerMethodField()
+    prev_push = serializers.SerializerMethodField()
+
+
+    def get_current_push(self, alert):
+        try:
+            datum_q = PerformanceDatum.objects.filter(
+                signature=alert.series_signature,
+                repository=alert.series_signature.repository,
+                push_id=alert.summary.push
+            )
+            if len(datum_q):
+                datum = datum_q.first()
+                metadata = TaskclusterMetadata.objects.get(job=datum.job)
+                return {
+                    'task_id': metadata.task_id,
+                    'retry_id': metadata.retry_id,
+                    'result': datum.job.result,
+                }
+            else:
+                return {}
+        except ObjectDoesNotExist:
+            return {}
+
+    def get_prev_push(self, alert):
+        try:
+            datum_q = PerformanceDatum.objects.filter(
+                signature=alert.series_signature,
+                repository=alert.series_signature.repository,
+                push_id=alert.summary.prev_push
+            )
+            if len(datum_q):
+                datum = datum_q.first()
+                metadata = TaskclusterMetadata.objects.get(job=datum.job)
+                return {
+                    'task_id': metadata.task_id,
+                    'retry_id': metadata.retry_id,
+                    'result': datum.job.result,
+                }
+            else:
+                return {}
+        except ObjectDoesNotExist:
+            return {}
+
+
+    class Meta:
+        model = Job
+        fields = [
+            # 'task_id',
+            # 'retry_id',
+            'current_push',
+            'prev_push',
+        ]
+
+
 class PerformanceAlertSerializer(serializers.ModelSerializer):
     series_signature = PerformanceSignatureSerializer(read_only=True)
+    taskcluster_metadata = serializers.SerializerMethodField()
```

Comment:
I'm wondering why it's not possible to use the serializer in the usual manner
```suggestion
    taskcluster_metadata = serializers.AlertTaskclusterMetadataSerializer()
```

Reply:
Rephrasing my question: Why don't you follow the similar approach of the other fields and create a custom serializer to encapsulate the logic there separately?